docs/types/mapping-types.rst

```diff
@@ -165,7 +165,8 @@ the ``sum`` function iterates over to sum all the values.
         function contains(itmap storage self, uint key) internal view returns (bool) {
             return self.data[key].keyIndex > 0;
         }
-
+        
+        // won't work with compiler 0.8 overflow checks
```

Comment:
That's a good spot, thank you very much :-). Rather than adding this comment we should probably fix the example to work properly with 0.8, though...

Reply:
On a very quick glance, a possible implementation would probably be something like
```
function iterate_start(itmap storage self) internal view returns (uint keyIndex) {
  if (0 < self.keys.length && self.keys[0].deleted)
            return iterate_next(self, 0);
  else
            return 0;
}
```
But might be worth spending some more thought on it.

Reply:
I thought about this as well, but there's that efficiency hit. Hate those one time checks!